### PR TITLE
chore: build the error reporting mechanism

### LINKED ISSUE #15396: Router - build the error reporting mechanism
Within this issue we should identify errors that we want to notify the client about, like:
- not enough token balance
- not enough ETH balance
- errors coming from packing the tx
- errors coming from estimate gas call
- other meaningful errors

Then build the system that will collect errors (potentially every skipped route (where we use `continue`) is skipped because of some error) that appeared during the loops in `resolveCandidates` function and at the end of the router algorithm if there are no routes we need to filter out those errors (cause will be probably a lot and many will be duplicated) and return to the client the list of errors sorted based on importance of errors.

### LINKED ISSUE #15693: Wrong value to send/receive are calculated when max value on chain is entered in SendModal
# Bug Report

## Description

Screenshot_2024-07-22_at_09 41 54

## Steps to reproduce

- Click Send
- enter max value available on chain (for me max on mainnet)
- when routes are calculated:
  - entered value is more than max value in top right corner
  - recipient wil get twice more than we send
  - value on mainnet is twice more than the balance on it

#### Expected behavior
Recipient should get the proper value (To be determined - is it sent - fee, or is it equal to sent?)
#### Actual behavior
See screenshot
### Additional Information

- Status desktop version: 2b415bdc5359f424fbfcabf1b5319bc3a8595921, status-go 94b20172262bede4ac4f9434c6e01e91a7868e1e
- Operating System: MacOS

### PR DESCRIPTION
**Pull request by saledjenic**
This PR brings `status-go` changes which close #15693
- https://github.com/status-im/status-go/pull/5615

and also introduces displaying routing errors in the UI, which closes #15396 

Closes #15396 #15693

How it was

image

Now

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/WalletUtils.qml [hunk 1]
```diff
@@ -130,4 +130,225 @@ QtObject {
...
+        case Constants.routerErrorCodes.processor.errBridgeHopCustomError:
+        case Constants.routerErrorCodes.processor.errBridgeCellerCustomError:
+        case Constants.routerErrorCodes.processor.errSwapParaswapCustomError:
+        case Constants.routerErrorCodes.processor.errENSRegisterCustomError:
+        case Constants.routerErrorCodes.processor.errENSReleaseCustomError:
+        case Constants.routerErrorCodes.processor.errENSPublicKeyCustomError:
+        case Constants.routerErrorCodes.processor.errStickersBuyCustomError:
+            return details
+        case Constants.routerErrorCodes.router.errENSRegisterRequiresUsernameAndPubKey:
+            return sTr("username and public key are required for registering ens name")
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/WalletUtils.qml:282]:
```suggestion
            return qsTr("username and public key are required for registering ens name")
```

Comment (caybro) [reply ui/app/AppLayouts/Wallet/WalletUtils.qml:282]: And everywhere below too

**Review by caybro — COMMENTED** [review]
LGTM otherwise

File: ui/app/AppLayouts/Wallet/WalletUtils.qml (hunk 1 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Wallet/WalletUtils.qml]: Thanks! Updated. :)

**Review by caybro — APPROVED** [review]
LGTM :+1:

**Review by belalshehab — APPROVED** [review]

**Pull request merged**